src/core/server/plugins/integration_tests/plugins_service.test.ts

```diff
@@ -0,0 +1,161 @@
+      version = 'some-version',
+      requiredPlugins = [],
+      optionalPlugins = [],
+      kibanaVersion = '7.0.0',
+      configPath = [path],
+      server = true,
+      ui = true,
+    }: {
+      path?: string;
+      disabled?: boolean;
+      version?: string;
+      requiredPlugins?: string[];
+      optionalPlugins?: string[];
+      kibanaVersion?: string;
+      configPath?: ConfigPath;
+      server?: boolean;
+      ui?: boolean;
+    }
+  ): PluginWrapper => {
+    return new PluginWrapper({
+      path,
+      manifest: {
+        id,
+        version,
+        configPath: `${configPath}${disabled ? '-disabled' : ''}`,
+        kibanaVersion,
+        requiredPlugins,
+        optionalPlugins,
+        server,
+        ui,
+      },
+      opaqueId: Symbol(id),
+      initializerContext: { logger } as any,
+    });
+  };
+
+  beforeEach(async () => {
+    mockPackage.raw = {
+      branch: 'feature-v1',
+      version: 'v1',
+      build: {
+        distributable: true,
+        number: 100,
+        sha: 'feature-v1-build-sha',
+      },
+    };
+
+    const env = Env.createDefault(getEnvOptions());
+    const config$ = new BehaviorSubject<Record<string, any>>({
+      plugins: {
+        initialize: true,
+      },
+    });
+    const rawConfigService = rawConfigServiceMock.create({ rawConfig$: config$ });
+    const configService = new ConfigService(rawConfigService, env, logger);
+    await configService.setSchema(config.path, config.schema);
+
+    pluginsService = new PluginsService({
+      coreId: Symbol('core'),
+      env,
+      logger,
+      configService,
+    });
+  });
+
+  it("properly resolves `getStartServices` in plugin's lifecycle", async () => {
+    expect.assertions(3);
+
+    const pluginPath = 'plugin-path';
+
+    mockDiscover.mockReturnValue({
+      error$: from([]),
+      plugin$: from([
+        createPlugin('plugin-id', {
+          path: pluginPath,
+          configPath: 'path',
+        }),
+      ]),
+    });
+
+    let startDependenciesResolved = false;
+
+    const pluginInitializer = () =>
+      ({
+        setup: async (core, deps) => {
+          core.getStartServices().then(([coreStart, pluginDeps]) => {
+            startDependenciesResolved = true;
+          });
+        },
+        start: async (core, deps) => {
+          await new Promise(resolve => setTimeout(resolve, 10));
+          expect(startDependenciesResolved).toBe(false);
+        },
+      } as Plugin);
+
+    jest.doMock(
+      join(pluginPath, 'server'),
+      () => ({
+        plugin: pluginInitializer,
+      }),
+      {
+        virtual: true,
+      }
+    );
+
+    await pluginsService.discover();
+
+    const setupDeps = coreMock.createInternalSetup();
+    await pluginsService.setup(setupDeps);
+
+    expect(startDependenciesResolved).toBe(false);
+
+    const startDeps = coreMock.createInternalStart();
+    await pluginsService.start(startDeps);
+
+    expect(startDependenciesResolved).toBe(true);
```

Comment:
It might be worth verifying that the resolved dependencies are the objects you expect them to be.